ignite/handlers/param_scheduler.py

```diff
@@ -183,47 +252,338 @@ def simulate_values(cls, num_events: int, **scheduler_kwargs: Any) -> List[List[
 
-            .. code-block:: python
+        .. code-block:: python
 
-                import matplotlib.pylab as plt
+            import matplotlib.pyplot as plt
+            import numpy as np
+
+            step_state_param_values = np.array(
+                StepStateParameterScheduler.simulate_values(
+                    num_events=20, param_name="step_scheduled_param", initial_value=10, gamma=0.99, step_size=5
+                )
+            )
+
+            plt.plot(step_state_param_values[:, 0], step_state_param_values[:, 1], label="learning rate")
+            plt.xlabel("events")
+            plt.ylabel("values")
+            plt.legend()
 
-                plt.figure(figsize=(10, 7))
-                LinearCyclicalScheduler.plot_values(num_events=50, param_name='lr',
-                                                    start_value=1e-1, end_value=1e-3, cycle_size=10))
         """
-        try:
-            import matplotlib.pylab as plt
-        except ImportError:
-            raise RuntimeError(
-                "This method requires matplotlib to be installed. "
-                "Please install it with command: \n pip install matplotlib"
+        keys_to_remove = ["save_history"]
+        for key in keys_to_remove:
+            if key in scheduler_kwargs:
+                del scheduler_kwargs[key]
+        values = []
+        scheduler = cls(save_history=False, **scheduler_kwargs)
+        engine = Engine(lambda e, b: None)
+        for i in range(num_events):
+            scheduler(engine=engine)
+            values.append([i, getattr(engine.state, scheduler_kwargs["param_name"])])
+        return values
+
+
+class LambdaStateParameterScheduler(StateParameterScheduler):
+    """Update a parameter during training by using a user defined function.
+        User defined function is taking an event index as input and returns parameter value.
+
+    Args:
+        lambda_fn: user defined parameter update function.
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    Examples:
+
+        .. code-block:: python
+
+            ...
+            engine = Engine(train_step)
+
+            initial_value = 10
+            gamma = 0.99
+
+            lambda_state_parameter_scheduler = LambdaStateParameterScheduler(
+                param_name="custom_scheduled_param",
+                lambda_fn=lambda event_index: initial_value * gamma ** (event_index % 9),
             )
 
-        values = cls.simulate_values(num_events=num_events, **scheduler_kwargs)
-        label = scheduler_kwargs.get("param_name", "learning rate")
-        ax = plt.plot([e for e, _ in values], [v for _, v in values], label=label)
-        plt.legend()
-        plt.grid(which="both")
-        return ax
+            lambda_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+            engine.run([0] * 8, max_epochs=2)
```

Comment:
Let's also add a basic parameter printing handler such that we see the value changes

Reply:
I've added a parameter printing handler in the example section of `LambdaStatePScheduler`